From 596fec579f148648d907d90be354d22234a3b2f4 Mon Sep 17 00:00:00 2001
From: Michele Bini <michele.bini@gmail.com>
Date: Tue, 3 Dec 2013 17:11:33 +0100
Subject: [PATCH] Protect global namespace.

For unclear reasons, this seems to introduce a huge (~30%) slowdown, at least in node.js.

---
 curve255.js | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/curve255.js b/curve255.js
index 8b4a48b..e8ed8ca 100644
--- a/curve255.js
+++ b/curve255.js
@@ -18,6 +18,8 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
+(function() {
+
 var c255lbase32chars = "abcdefghijklmnopqrstuvwxyz234567";
 var c255lbase32values = {"a":0, "b":1, "c":2, "d":3, "e":4, "f":5, "g":6, "h":7, "i":8, "j":9, "k":10, "l":11, "m":12, "n":13, "o":14, "p":15, "q":16, "r":17, "s":18, "t":19, "u":20, "v":21, "w":22, "x":23, "y":24, "z":25, "2":26, "3":27, "4":28, "5":29, "6":30, "7":31 };
 function c255lbase32encode(n, x) {
@@ -452,3 +454,13 @@ function curve25519(f, c) {
     f[15]  = (f[15] & 0x7FFF) | 0x4000;
     return curve25519_raw(f, c);
 }
+
+this.c255lzero       = c255lzero;
+this.c255lhexdecode  = c255lhexdecode;
+this.c255lhexencode  = c255lhexencode;
+
+this.c255lbigintcmp = c255lbigintcmp;
+this.curve25519_raw = curve25519_raw;
+this.curve25519 = curve25519;
+    
+})(this);
-- 
1.8.4.rc3

